### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by dabonnie**
* Add lint action
* Add test action

Signed-off-by: Devin Bonnie <dbbonnie@amazon.com>

### CODE REVIEW

**Comment by thomas-moulard** [comment]
The failling linters will not fail after you add code, so maybe you can merge this after #2? Or remove the linters until #2 is merged.

**Comment by dabonnie** [comment]
> The failling linters will not fail after you add code, so maybe you can merge this after #2? Or remove the linters until #2 is merged.

There are a few complaints of missing packages for a few linters: maybe we can sit down and take a look at those.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+  name: Test file_talker
```
Comment (dabonnie) [inline .github/workflows/test.yml]: Todo remove the two spaces

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+on:
+  pull_request:
+
+jobs:
+  ament_lint:
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+      matrix:
+        linter: [copyright, cppcheck, cpplint, flake8, lint_cmake, linter, pep257, uncrustify, xmllint]
```
Comment (dabonnie) [inline .github/workflows/lint.yml]:
```suggestion
        linter: [copyright, cppcheck, cpplint, flake8, pep257, uncrustify, xmllint]
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (dabonnie) [inline .github/workflows/test.yml]:
```suggestion
name: Test file_talker
```

**Comment by dabonnie** [comment]
Linting now passing, for some reason tests are failing.

**Comment by thomas-moulard** [comment]
```
Starting >>> file_talker
--- output: file_talker
CMake Error: The source directory "/home/runner/work/file_talker/file_talker/ros2_ws/src/file_talker/file_talker" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
---
--- stderr: file_talker
CMake Error: The source directory "/home/runner/work/file_talker/file_talker/ros2_ws/src/file_talker/file_talker" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
---
```

You don't have a CMakeLists. Did you rebase?

**Comment by dabonnie** [comment]
Weird, the file exists in master: https://github.com/ros-tooling/file_talker/blob/master/file_talker/CMakeLists.txt

**Comment by dabonnie** [comment]
> ```
> Starting >>> file_talker
> --- output: file_talker
> CMake Error: The source directory "/home/runner/work/file_talker/file_talker/ros2_ws/src/file_talker/file_talker" does not appear to contain CMakeLists.txt.
> Specify --help for usage, or press the help button on the CMake GUI.
> ---
> --- stderr: file_talker
> CMake Error: The source directory "/home/runner/work/file_talker/file_talker/ros2_ws/src/file_talker/file_talker" does not appear to contain CMakeLists.txt.
> Specify --help for usage, or press the help button on the CMake GUI.
> ---
> ```
> 
> You don't have a CMakeLists. Did you rebase?

Didn't rebase, it exists in master: 
image

**Comment by thomas-moulard** [comment]
Yep, so that won't work, it's testing what you have on *this* branch. You'll need to run`git-merge` or `git-rebase`.

**Comment by thomas-moulard** [comment]
@dabonnie the OS X bug is on my side, could you disable it for now? (remove it from the os list). This is unrelated to your package.

**Pull request merged**